ignitionrobotics/ign-gazebo
# Model editor: Add links to model

# 🎉 New feature

## Summary

This pull request adds a `+` menu button to the Component Inspector for adding links to models. Currently supports adding links consisting of simple shapes and lights.

The changes are written to the ECM and GUI side and assumes that GUI and server ECM synchronization (#1109) is available. Marked as draft until the newly added entities can be reflected on the server side.

## Test it

1. launch ign gazebo with shapes.sdf world
1. Open the Model Editor GUI plugin by going to top right 3 dots menu and select `Model Editor`
1. Click on one of the shapes, e.g. box
1. Go the Component Inspect and you should see a new `+` button appear on the top bar
1. Click on the `+` button and select a link type to add, .e.g Cylinder
1. Once selected, a black cylinder should now appear at where the original box model is.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
